django/templatetags/tz.py

```diff
@@ -1,13 +1,35 @@
 from datetime import datetime, tzinfo
 
-import pytz
+try:
+    import zoneinfo
+except ImportError:
+    from backports import zoneinfo
 
+from django.conf import settings
 from django.template import Library, Node, TemplateSyntaxError
 from django.utils import timezone
 
 register = Library()
 
 
+class UnknownTimezoneException(BaseException):
+    pass
+
+
+# RemovedInDjango50Warning
+def timezone_constructor(tzname):
+    if settings.USE_DEPRECATED_PYTZ:
+        import pytz
+        try:
+            return pytz.timezone(tzname)
+        except pytz.UnknownTimeZoneError:
+            raise UnknownTimezoneException()
```

Comment:
An exception doesn't need to been instantiated: 
```suggestion
            raise UnknownTimezoneException
```